kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/records/BatchAwareMemoryRecordsBuilder.java

```diff
@@ -144,6 +145,21 @@ public BatchAwareMemoryRecordsBuilder addBatch(CompressionType compressionType,
                 templateBatch.deleteHorizonMs().orElse(RecordBatch.NO_TIMESTAMP));
     }
 
+    /**
+     * Directly appends a batch, intended to be used for passing through unmodified batches. Writes
+     * the previous batch to the stream if required
```

Comment:
Should the javadoc say the batch will be closed too?

Reply:
which batch? the previous batch?